azurerm/resource_arm_sql_server.go

```diff
@@ -60,101 +59,111 @@ func resourceArmSqlServer() *schema.Resource {
+	}
+
+	tags := d.Get("tags").(map[string]interface{})
+	metadata := expandTags(tags)
+
+	props := sql.ServerProperties{
+		FullyQualifiedDomainName:   &fqdn,
+		Version:                    serverVersion,
+		AdministratorLogin:         &admin,
+		AdministratorLoginPassword: &adminPw,
 	}
 
-	readRequest := rivieraClient.NewRequest()
-	readRequest.Command = &sql.GetServer{
-		Name:              d.Get("name").(string),
-		ResourceGroupName: d.Get("resource_group_name").(string),
+	parameters := sql.Server{
+		Name:             &name,
+		ServerProperties: &props,
+		Tags:             metadata,
+		Location:         &location,
 	}
 
-	readResponse, err := readRequest.Execute()
+	//last parameter is set to empty to allow updates to records after creation
+	// (per SDK, set it to '*' to prevent updates, all other values are ignored)
+	result, err := sqlServersClient.CreateOrUpdate(resGroup, name, parameters)
 	if err != nil {
-		return fmt.Errorf("Error reading SQL Server: %s", err)
+		return err
 	}
-	if !readResponse.IsSuccessful() {
-		return fmt.Errorf("Error reading SQL Server: %s", readResponse.Error)
+
+	if result.ID == nil {
+		return fmt.Errorf("Cannot create Sql Server %s (resource group %s) ID", name, resGroup)
 	}
 
-	resp := readResponse.Parsed.(*sql.GetServerResponse)
-	d.SetId(*resp.ID)
+	d.SetId(*result.ID)
 
 	return resourceArmSqlServerRead(d, meta)
 }
 
 func resourceArmSqlServerRead(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
+	sqlServersClient := meta.(*ArmClient).sqlServersClient
 
 	id, err := parseAzureResourceID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	readRequest := rivieraClient.NewRequestForURI(d.Id())
-	readRequest.Command = &sql.GetServer{}
+	resGroup := id.ResourceGroup
+	name := id.Path["servers"]
 
-	readResponse, err := readRequest.Execute()
+	result, err := sqlServersClient.Get(resGroup, name)
 	if err != nil {
-		return fmt.Errorf("Error reading SQL Server: %s", err)
+		return fmt.Errorf("Error reading SQL Server %s: %v", name, err)
 	}
-	if !readResponse.IsSuccessful() {
-		log.Printf("[INFO] Error reading SQL Server %q - removing from state", d.Id())
+	if result.Response.StatusCode == http.StatusNotFound {
 		d.SetId("")
-		return fmt.Errorf("Error reading SQL Server: %s", readResponse.Error)
+		return nil
 	}
 
-	resp := readResponse.Parsed.(*sql.GetServerResponse)
+	serverProperties := *result.ServerProperties
 
-	d.Set("name", id.Path["servers"])
-	d.Set("resource_group_name", id.ResourceGroup)
-	d.Set("location", azureRMNormalizeLocation(*resp.Location))
-	d.Set("fully_qualified_domain_name", resp.FullyQualifiedDomainName)
-	d.Set("administrator_login", resp.AdministratorLogin)
-	d.Set("version", resp.Version)
+	serverVersion := serverProperties.Version
 
-	flattenAndSetTags(d, resp.Tags)
+	d.Set("name", name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("version", string(serverVersion))
+	d.Set("location", *result.Location)
```

Comment:
can we update `location` to use the `azureRMNormalizeLocation(*resp.Location)` method? this helps ensure it's stored consistently in the statefile